Frontend/src/components/SearchResults.js

```diff
@@ -1,17 +1,23 @@
 import React from 'react'
 
-const SearchResults = ( {results} ) => {
+const SearchResults = ({ results, onStockSelected }) => {
   return (
-    <ul className='absolute top-12 border-2 w-full rounded-md h-64 overflow-y-scroll bg-white border-neutral-200'>
-        {results.map((item) =>{
-            // console.log('Key:', item["1. symbol"]); // Log the key
-            return <li key={item["1. symbol"]} className='cursor-pointer p-4 m-2 flex items-center justify-between rounded-md hover:bg-emerald-200'>
-                <span>{item["1. symbol"]}</span>
-                <span>{item["2. name"].substring(0, 25)}</span>
-            </li>
-        })}
-    </ul>
-  )
+      <ul className='absolute top-12 border-2 w-full rounded-md h-64 overflow-y-scroll bg-white border-neutral-200'>
+          {results.map((item) => {
+              return (
+                  <li 
+                      key={item["1. symbol"]} 
+                      className='cursor-pointer p-4 m-2 flex items-center justify-between rounded-md hover:bg-emerald-200'
+                      onClick={() => onStockSelected(item["1. symbol"])}
+                  >
+                      <span>{item["1. symbol"]}</span>
+                      <span>{item["2. name"].substring(0, 25)}</span>
```

Comment:
Much easier to have a transformation layer and the API can change/update to a different API with minimal effort because all you have to do is change the transformation layer and your whole app doesn't have to change.  Think like this, developers are lazy, so we work smarter, not harder per say.